assets/src/edit-story/components/dropTargets/provider.js

```diff
@@ -0,0 +1,144 @@
+/*
+ * Copyright 2020 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * External dependencies
+ */
+import PropTypes from 'prop-types';
+import { throttle } from 'throttle-debounce';
+
+/**
+ * WordPress dependencies
+ */
+import { useState, useCallback } from '@wordpress/element';
+
+/**
+ * Internal dependencies
+ */
+import { mergeElements } from '../../elements';
+import { useTransform } from '../transform';
+import { useStory } from '../../app';
+import Context from './context';
+
+function DropTargetsProvider({ children }) {
+  const [dropTargets, setDropTargets] = useState([]);
+  const [activeDropTarget, setActiveDropTarget] = useState(null);
+  const {
+    actions: { pushTransform },
+  } = useTransform();
+  const {
+    actions: { deleteSelectedElements, updateElementById },
+    state: { currentPage },
+  } = useStory();
+
+  const getDropTargetFromCursor = useCallback(
+    (x, y, element) => {
+      const underCursor = document.elementsFromPoint(x, y);
+      return (
+        Object.keys(dropTargets).find(
+          (id) => underCursor.includes(dropTargets[id]) && id !== element.id
```

Comment:
dropTargets follow the z-indexing of the frame elements which in this case works fine

Reply:
Note: based on testing it does seem to highlight the incorrect shape sometimes with multiple shapes largely on top of each other, however, it might be from the issue that detecting being under cursor doesn't seem to work well with the "circle" shape too well -- it seems to detect the shape only when moving cursor to the edge of the circle.